Shared/sdk/CScriptArgReader.h

```diff
@@ -1007,6 +1007,25 @@ class CScriptArgReader
         m_iIndex++;
     }
 
+    //
+    // Reads a table of string key-value pairs
+    //
+    void ReadPairTable(std::vector<std::pair<SString, SString>>& outPairs)
+    {
+        outPairs.clear();
+
+        int argument = lua_type(m_luaVM, m_iIndex);
+        if (argument == LUA_TTABLE)
+        {
+            InternalReadPair(outPairs, m_iIndex);
+            ++m_iIndex;
+            return;
+        }
+
+        SetTypeError("table");
+        ++m_iIndex;
+    }
+
```

Comment:
Every time I see a vector, I want to use `reserve()`
2 strings are 16 bytes, so if you reserve like 128, then `shrink_to_fit()` you should be good.
Also, you could just use `lua_rawlen`, which may or may not return the correct value. [See here](https://stackoverflow.com/questions/26643285/get-lua-table-size-in-c) for more info